lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1531,7 +1531,8 @@ func (d *zfs) ValidateVolume(vol Volume, removeUnknownKeys bool) error {
 	// when using custom filesystem volumes with block mode enabled. LXD will create the filesystem
 	// for these volumes, and use the mount options. When attaching a regular block volumes to a VM,
 	// these are not mounted by LXD and therefore don't need these config keys.
-	if vol.volType == VolumeTypeCustom && (vol.contentType == ContentTypeBlock || !vol.IsBlockBacked()) {
+	if vol.IsVMBlock() || vol.volType == VolumeTypeCustom && (vol.contentType == ContentTypeBlock || !vol.IsBlockBacked()) {
```

Comment:
Do we need brackets around `vol.volType == VolumeTypeCustom && (vol.contentType == ContentTypeBlock || !vol.IsBlockBacked())`?

Reply:
No, we don't. Will fix.

Reply:
No I meant it should be like this right?

```
|| (vol.volType == VolumeTypeCustom && (vol.contentType == ContentTypeBlock || !vol.IsBlockBacked()))
```

Reply:
Sorry, yes we do - since we check `vol.contentType` and `!vol.IsBlockBacked` only for custom volumes and if either one of them is true, we should remove block configurations.

Reply:
I see, I see. Yes, I can update this.